src/include/wiredtiger.in

```diff
@@ -4021,6 +4021,51 @@ const char *wiredtiger_version(int *majorp, int *minorp, int *patchp)
  */
 /*! @} */
 
+/*******************************************
+ * Sub-level error returns
+ *******************************************/
+/*!
+ * @name Sub-level error returns
+ * Along with error returns, WiredTiger can return an integer code indicating
+ * a more specific error. A return of -32,000 indicates that there is no more
+ * specific error; all other return values indicate a more specific error which
+ * provides greater context into the failure.
+ *
+ * WiredTiger reserves all values from -32,000 to -31,199 as possible sub-level
+ * error return values.
+ *
+ * The following are all of the WiredTiger-specific sub-level error returns:
+ * @{
+ */
+/*
+ * DO NOT EDIT: automatically built by dist/api_err.py.
+ * Sub-level error return section: BEGIN
+ */
+/*!
+ * Last API call was successful.
+ * This is the default sub-level error code that should be used when there is no
+ * sub-level error to pair with an error. It indicates that no further context
+ * exists or is necessary.
+ */
+#define	WT_NONE	(-32000)
+/*!
+ * Cannot reconfigure background compaction while it's already running.
+ * This error is generated when the user tries to reconfigure the background
+ * compaction while it is already running.
+ */
+#define	WT_COMPACTION_ALREADY_RUNNING	(-32001)
+/*!
+ * Out of sessions, configured for XXX (including internal sessions).
```

Comment:
Is the XXX expected?

Reply:
I'm not sure if it should be replaced by something else later, but that was the message on the tech design, same for the other sub-level error code messages.

Reply:
XXX would be there as a placeholder - let's see if we can improve on it...

Reply:
We'll need some smarts to handle the XXX, as that is suggested to the number of actual configured sessions.
For now, I suggest going with:
`"out of sessions (including internal sessions)"`
for that message and we can work on the integer number of sessions later.

Reply:
Fixed